dangerfile.ts

```diff
@@ -1,71 +1,139 @@
+    warn(`${message} - <em>${idea}</em>`)
+  }
 
-// Request update of yarn.lock if package.json changed but yarn.lock isn't
-const packageChanged = includes(allFiles, 'package.json')
-const lockfileChanged = includes(allFiles, 'yarn.lock')
-if (packageChanged && !lockfileChanged) {
-  const message = 'Changes were made to package.json, but not to yarn.lock'
-  const idea = 'Perhaps you need to run `yarn install`?'
-  warn(`${message} - <i>${idea}</i>`)
-}
+  // Require new src/components files to include changes to storybook
+  const hasStorybookChanges = allFiles.some(
+    (p) => !!p.match(/src\/.*\.stories\.[jt]sx?/)
+  )
 
-// Ensure we have access to github for these checks
-let isYarnAuditMissing = false
-if (danger.github) {
-  const prBody = danger.github.pr.body
-  if (lockfileChanged && danger.github.pr.user.type == 'User') {
-    isYarnAuditMissing = !(
-      includes(prBody, 'vulnerabilities found') &&
-      includes(prBody, 'Packages audited:')
+  if (hasCodeChanges && !hasStorybookChanges) {
+    warn(
+      'This PR does not include changes to storybook, even though it affects component code.'
     )
   }
 }
 
-// Encourage adding `yarn audit` output on package change
-if (isYarnAuditMissing) {
-  const message =
-    'Changes were made to yarn.lock, but no plain text yarn audit output was found in PR description.'
-  const idea =
-    'Can you run `yarn audit` in your branch and paste the results inside a markdown code block?'
-  warn(`${message} - <i>${idea}</i>`)
+const checkDependencyChanges: () => void = () => {
+  // Request update of yarn.lock if package.json changed but yarn.lock isn't
+  const packageChanged = allFiles.includes('package.json')
+  const lockfileChanged = allFiles.includes('yarn.lock')
+  if (packageChanged && !lockfileChanged) {
+    danger.git
+      .structuredDiffForFile('package.json')
+      .then((sdiff) => {
+        return sdiff.chunks.every((chunk) => {
+          return chunk.changes
+            .filter((change) => {
+              // filter out changes that are context lines in the diff
+              return change.type !== 'normal'
+            })
+            .every((change) => {
+              // for every add/del, is the only change to the version?
+              return change.content.match(/"version":/)
+            })
+        })
+      })
+      .then((onlyVersionChanges) => {
+        // If the only thing that changed is the version, it is ok if
+        // yarn.lock didn't change
+        if (!onlyVersionChanges) {
+          const message =
+            'Changes were made to package.json, but not to yarn.lock'
+          const idea = 'Perhaps you need to run `yarn install`?'
+          warn(`${message} - <i>${idea}</i>`)
+        }
+      })
+  }
+}
+
+// skip these checks if PR is by dependabot, if we don't have a github object let it run also since we are local
+if (
+  !danger.github ||
+  (danger.github && danger.github.pr.user.login !== 'dependabot-preview[bot]')
```

Comment:
In other places (circleci config) we check for 'dependabot-preview[bot]' and `dependabot[bot]' . Assuming that's no longer needed? Just checking

Reply:
Not sure.  I'll try to check

Reply:
I *think* I found a way to detect all bots, so I think that will do what we want

Reply:
Thanks for looking for into it 🚀